Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -1247,20 +1247,51 @@ static GstPadProbeReturn appendPipelinePadProbeDebugInformation(GstPad*, GstPadP
 #endif
 
 #if ENABLE(ENCRYPTED_MEDIA)
-static GstPadProbeReturn appendPipelineAppsinkPadEventProbe(GstPad*, GstPadProbeInfo* info, struct PadProbeInformation *padProbeInformation)
+void AppendPipeline::handleProtectedBufferProbeInformation(GstPadProbeInfo* info)
+{
+    if (!m_cachedProtectionEvents.size())
+        return;
+
+    GstBuffer* buffer = gst_pad_probe_info_get_buffer(info);
+    GST_DEBUG("adding %u protection events to buffer %p", m_cachedProtectionEvents.size(), buffer);
+
+    GValue* streamEncryptionEventsList = g_new0(GValue, 1);
+    g_value_init(streamEncryptionEventsList, GST_TYPE_LIST);
+    for (auto& event : m_cachedProtectionEvents) {
```

Comment:
I'm a little uncertain if all the cached protection event we keep here and then later attach to an outgoing buffer are guaranteed to be associated with that buffer, is the ordering of these events / buffers guaranteed by gstreamer?

Reply:
Yes, since they are regular downstream events (non OOB).